test/logflare/backends/user_monitoring_test.exs

```diff
@@ -0,0 +1,258 @@
+        user.id
+        |> Sources.create_user_system_sources()
+        |> Enum.find(&(&1.system_source_type == :metrics))
+
+      start_supervised!({SourceSup, system_source})
+
+      :telemetry.execute([:logflare, :test, :user_specific], %{value: 456}, metadata)
+
+      user_exporter_metrics =
+        OtelMetricExporter.MetricStore.get_metrics(:"system.metrics-#{user.id}")
+
+      refute match?(
+               %{
+                 {:last_value, "logflare.test.user_specific.value"} => %{
+                   %{backend_id: ^backend_id} => 456
+                 }
+               },
+               user_exporter_metrics
+             )
+    end
+
+    test "dont get mixed between users", %{
+      user_1: user_1,
+      user_2: user_2,
+      backend_1: %{id: backend_1_id},
+      backend_2: %{id: backend_2_id}
+    } do
+      # user 1 setup
+      user_1 |> Users.update_user_allowed(%{system_monitoring: true})
+
+      system_source_1 =
+        Sources.get_by(user_id: user_1.id, system_source_type: :metrics)
+
+      start_supervised!({SourceSup, system_source_1}, id: {:source_sup, 1})
+
+      # user 2 setup
+
+      user_2 |> Users.update_user_allowed(%{system_monitoring: true})
+
+      system_source_2 =
+        Sources.get_by(user_id: user_2.id, system_source_type: :metrics)
+
+      start_supervised!({SourceSup, system_source_2}, id: {:source_sup, 2})
+
+      # sending signals related to both users
+
+      :telemetry.execute([:logflare, :test, :user_specific], %{value: 123}, %{
+        backend_id: backend_1_id
+      })
+
+      :telemetry.execute([:logflare, :test, :user_specific], %{value: 456}, %{
+        backend_id: backend_2_id
+      })
+
+      user_1_exporter_metrics =
+        OtelMetricExporter.MetricStore.get_metrics(:"system.metrics-#{user_1.id}")
+
+      assert match?(
+               %{
+                 {:last_value, "logflare.test.user_specific.value"} => %{
+                   %{backend_id: ^backend_1_id} => 123
+                 }
+               },
+               user_1_exporter_metrics
+             )
+
+      refute match?(
+               %{
+                 {:last_value, "logflare.test.user_specific.value"} => %{
+                   %{backend_id: ^backend_2_id} => _
+                 }
+               },
+               user_1_exporter_metrics
+             )
+
+      user_2_exporter_metrics =
+        OtelMetricExporter.MetricStore.get_metrics(:"system.metrics-#{user_2.id}")
+
+      assert match?(
+               %{
+                 {:last_value, "logflare.test.user_specific.value"} => %{
+                   %{backend_id: ^backend_2_id} => 456
+                 }
+               },
+               user_2_exporter_metrics
+             )
+
+      refute match?(
+               %{
+                 {:last_value, "logflare.test.user_specific.value"} => %{
+                   %{backend_id: ^backend_1_id} => _
+                 }
+               },
+               user_2_exporter_metrics
+             )
```

Comment:
instead of testing this, shouldn't we instead assert that the metrics ingested into the system sources?
reaching into OtelMetricsExporter storage doesn't cover the callback areas.